pkg/updater/service/datadog_agent.go

```diff
@@ -72,15 +72,16 @@ func SetupAgentUnits() (err error) {
 			return
 		}
 	}
+	// write installinfo before start, or the agent could write it
+	if err = installinfo.WriteInstallInfo("updater_package", "manual_update"); err != nil {
+		return
+	}
 	for _, unit := range stableUnits {
 		if err = startUnit(unit); err != nil {
 			return
 		}
 	}
-	if err = createAgentSymlink(); err != nil {
-		return
-	}
-	err = installinfo.WriteInstallInfo("updater_package", "manual_update_via_apt")
+	err = createAgentSymlink()
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Why is this error not handled? Is the linter running on `pkg/updater`? I am surprised this is passing

Reply:
the err still passes in the return, thanks to the func signature
`SetupAgentUnits() (err error)`